nodeup/pkg/model/kubelet.go

```diff
@@ -771,16 +768,28 @@ func (b *KubeletBuilder) kubeletNames() ([]string, error) {
 		return append(addrs, name), nil
 	}
 
-	cloud := b.Cloud.(awsup.AWSCloud)
+	addrs := []string{b.InstanceID}
+	sess := session.Must(session.NewSession())
+	metadata := ec2metadata.New(sess)
 
-	result, err := cloud.EC2().DescribeInstances(&ec2.DescribeInstancesInput{
-		InstanceIds: []*string{&b.InstanceID},
-	})
-	if err != nil {
-		return nil, fmt.Errorf("error describing instances: %v", err)
+	if localHostname, err := metadata.GetMetadata("local-hostname"); err != nil {
```

Comment:
```suggestion
	if localHostname, err := metadata.GetMetadata("local-hostname"); err == nil {
```

Reply:
Are you reviewing your own PRs? :-)

Reply:
away from laptop and thought I can fix these nits, though not easy to bulk commit 😄